test/built-ins/RegExp/prototype/Symbol.search/set-lastindex-restore.js

```diff
@@ -6,25 +6,29 @@ es6id: 21.2.5.9
 description: The `lastIndex` value is restored following match execution
 info: >
     [...]
-    11. Let status be Set(rx, "lastIndex", previousLastIndex, true).
+    8. If SameValue(currentLastIndex, previousLastIndex) is false, then
+       a. Perform ? Set(rx, "lastIndex", previousLastIndex, true).
     [...]
 features: [Symbol.search]
 ---*/
 
-var latestValue;
+var latestValue = 86;
+var callCount = 0;
 var fakeRe = {
   get lastIndex() {
-    return 86;
+    return latestValue;
```

Comment:
I think the test is working as intended, but it's quite subtle:

1. Symbol.search sets lastIndex to 0 prior to the exec call (since previousLastIndex != 0).
2. exec sets lastIndex to null.
3. Symbol.search restores lastIndex to 86 (since previousLastIndex != currentLastIndex).

We check that exec was actually called, and that the value was restored to 86 after the its call.